src/test/java/com/google/devtools/build/lib/remote/RemoteSpawnRunnerTest.java

```diff
@@ -356,8 +352,93 @@ public void dontAcceptFailedCachedAction() throws Exception {
                 retrier,
                 digestUtil,
                 logDir));
+    Spawn spawn = newSimpleSpawn();
+    SpawnExecutionContext policy = new FakeSpawnExecutionContext(spawn);
+
+    SpawnResult res = Mockito.mock(SpawnResult.class);
+    when(res.exitCode()).thenReturn(1);
+    when(res.status()).thenReturn(Status.EXECUTION_FAILED);
+    when(localRunner.exec(eq(spawn), eq(policy))).thenReturn(res);
 
-    assertThrows(EnvironmentalExecException.class, () -> runner.exec(spawn, policy));
+    runner.exec(spawn, policy);
+
+    verify(localRunner).exec(eq(spawn), eq(policy));
+    verify(runner)
+        .execLocallyAndUpload(
+            eq(spawn),
+            eq(policy),
+            any(SortedMap.class),
+            eq(cache),
+            any(ActionKey.class),
+            any(Action.class),
+            any(Command.class),
+            /* uploadLocalResults= */ eq(true));
+    verify(cache, never())
```

Comment:
upload should be called because `uploadLocalResults` is `true`. Generally, if a failed action was cached which then succeeds locally we want to overwrite the failed result with the successful result in the cache. This would typically be used for flaky tests.

Reply:
`when(res.exitCode()).thenReturn(1);
    when(res.status()).thenReturn(Status.EXECUTION_FAILED);`
because of these 2 lines the local action also failed => no upload expected to the cache.

But I think successful use case makes more sense here, so changed to cache miss + successful local execution => cache.upload(...)